rfc/rfc-79/rfc-79.md

```diff
@@ -0,0 +1,140 @@
+
+The new /.cancel folder will enable goal (B) by allowing writers to permentantly prevent an ongoing cancelltable table service write from comitting, without needing to block/wait for the table service writer (or rely on a conflicting ingestion write appearing and taking precedence). The new ".aborted" state will allow writers to infer wether a cancelled table service plan needs to still have it's partial data writes cleaned up from the dataset, which is needed for (C). The additional design change below will complete the remaining requirement for (C) of eventual cleanup.
+  
+### Handling cancellation of plans
+An additional config "cancellation-policy" can be added to the table service plan to indicate when it is ellgible to be permanently cancelled by writers other than the one responsbible for executing the table service. This policy can be a threshold of hours or instants on timeline, where if that # of hours/instants have elapsed since the plan was scheduled, any call to clean will cleanup the instant. This policy should be configured by the writer scheduling a cancellable table service, based on the amount of time they expect the plan to remain on the timeline before being picked up for execution. For example, if the plan is expected to have its execution deferred to a few hours later, then the cancellation-policy should be lenient in allowing the plan to remain many hours on the timeline before being subject to clean's cancellation. Note that this cancellation policy is not a repalacement for determining wether a table service plan is currently being executed - as with ingestion writes, permanent cleanup of a cancellable table service plan will only start once it is confirmed that a ongoing writer is no longer progressing it. 
+
+In order to ensure that other writers can indeed permenantely cancel a cancellable table service plan (such that it can no longer be executed), additional changes to clean and table service flow will be need to be added as well, as will be proposed below. Also, note that the cancellation-policy is only required to be honored by clean: a user can choose setup an application to aggresively clean up a failed cancellable table service plan even if it has not meet the critera for its cancellation-policy yet. This can be useful if a user wants a utility to manually ensure that clean/archival for a dataset progresses immdeitately or knows that a cancellable table service plan will not be attempted again or cleaned up by another writer. The two new cancel APIs in the below propsoal provide a method to achieve this.
+  
+
+#### Enabling table service execution and clean to support cancellation and automatic cleanup
+
```

Comment:
this is regarding rolling back a "cancelled" clustering. 
wondering at the end of clustering, if we detect that there is a cancellation request, can we just abort the transaction. 
so next time when we resume, first thing we will check if there is any cancel and then trigger rollback and abort. 
that way, rollbacks for clustering is always triggered at one place in the code for a table service and not at many diff places.